sys/net/application_layer/nanocoap/sock.c

```diff
@@ -78,6 +108,11 @@ ssize_t nanocoap_request(coap_pkt_t *pkt, sock_udp_ep_t *local, sock_udp_ep_t *r
                 DEBUG("nanocoap: error parsing packet\n");
                 res = -EBADMSG;
             }
+            else if (coap_get_id(pkt) != id) {
+                res = -EBADMSG;
+                continue;
+            }
```

Comment:
Does it make sense not to decrement the retry-count if you got a 'bad' message?

Reply:
Yes…

Reply:
Oh I see you only re-transmit the message if `res == -EAGAIN` and you just set it to `-EBADMSG` here.